rllib/agents/ppo/ppo_torch_policy.py

```diff
@@ -187,11 +188,14 @@ def __init__(self, obs_space, action_space, config):
 
             def value(ob, prev_action, prev_reward, *state):
                 model_out, _ = self.model({
-                    SampleBatch.CUR_OBS: torch.Tensor([ob]),
-                    SampleBatch.PREV_ACTIONS: torch.Tensor([prev_action]),
-                    SampleBatch.PREV_REWARDS: torch.Tensor([prev_reward]),
+                    SampleBatch.CUR_OBS: torch.Tensor([ob]).to(self.device),
+                    SampleBatch.PREV_ACTIONS: torch.Tensor([prev_action]).to(
+                        self.device),
+                    SampleBatch.PREV_REWARDS: torch.Tensor([prev_reward]).to(
+                        self.device),
```

Comment:
@sven1977 I had to add a bunch of these to get it to run on a GPU. Not sure if there's a way we can test this easily, or if this is an unfortunate pytorch fact of life.

Reply:
Yeah, let's see what these all do on CPU vs GPU machines. It is annoying to have to do this in torch. Noticed that as well, when I translated PPO in the first place. I'm too suspecting that somewhere in either the ".cpu()" or the ".to()"-calls lies our PPO torch memory leak. :/

Reply:
We could potentially have interceptors in TorchPolicy that do this conversion on all tensor outputs, to avoid forgetting. I think it would have to cover the extra action outs, extra grad outs, any preprocessor outputs, and the sampler too (probably forgetting some). But that can happen in another PR.